pywebhdfs/webhdfs.py

```diff
@@ -77,8 +83,17 @@ def create_file(self, path, file_data, **kwargs):
 
         # make the initial CREATE call to the HDFS namenode
         optional_args = kwargs
-        uri = self._create_uri(path, operations.CREATE, **optional_args)
-        init_response = requests.put(uri, allow_redirects=False)
+
+        hosts = self._resolve_federation(path)
+        for host in hosts:
+            uri = self._create_uri(host, path, operations.CREATE,
+                                   **optional_args)
+            init_response = requests.put(uri, allow_redirects=False)
+            if not _is_standby_exception(init_response):
```

Comment:
not for this PR, but I'm thinking a log handler really needs to be added so that we can emit a log message when the namenode is failing over